src/libraries/System.Net.Http/src/System/Net/Http/Headers/HeaderDescriptor.cs

```diff
@@ -148,5 +166,110 @@ private static bool TryDecodeUtf8(ReadOnlySpan<byte> input, out string decoded)
             decoded = null;
             return false;
         }
+
+        // QPack Static Table
+        // https://tools.ietf.org/html/draft-ietf-quic-qpack-11#appendix-A
+        private static readonly (HeaderDescriptor descriptor, string value)[] s_qpackHeaderLookup = new (HeaderDescriptor descriptor, string value)[]
```

Comment:
Why we need this here? Seems like a lot of duplication with QPackStaticTable.cs.

Reply:
I can probably generate this at runtime.

Reply:
It's an optimization.

Generating at runtime left as TODO; @Tratcher has some ideas on abstracting this.